src/pubsub.js

```diff
@@ -1,48 +1,70 @@
-
 'use strict';
 
 var nconf = require('nconf');
-var util = require('util');
-var winston = require('winston');
-var EventEmitter = require('events').EventEmitter;
-
-var channelName;
-
-var PubSub = function () {
-	var self = this;
-	if (nconf.get('redis')) {
-		var redis = require('./database/redis');
-		var subClient = redis.connect();
-		this.pubClient = redis.connect();
-
-		channelName = 'db:' + nconf.get('redis:database') + 'pubsub_channel';
-		subClient.subscribe(channelName);
-
-		subClient.on('message', function (channel, message) {
-			if (channel !== channelName) {
-				return;
-			}
-
-			try {
-				var msg = JSON.parse(message);
-				self.emit(msg.event, msg.data);
-			} catch (err) {
-				winston.error(err.stack);
-			}
-		});
-	}
+
+var real;
+var fake = {
+	publishQueue: [],
+	publish: function (event, data) {
+		fake.publishQueue.push({ event: event, data: data });
+	},
+	listenQueue: {},
+	on: function (event, callback) {
+		if (!Object.prototype.hasOwnProperty.call(fake.listenQueue, event)) {
+			fake.listenQueue[event] = [];
+		}
+		fake.listenQueue[event].push(callback);
+	},
+	removeAllListeners: function (event) {
+		delete fake.listenQueue[event];
+	},
 };
 
-util.inherits(PubSub, EventEmitter);
+function get() {
+	if (real) {
+		return real;
+	}
+
+	var pubsub;
 
-PubSub.prototype.publish = function (event, data) {
-	if (this.pubClient) {
-		this.pubClient.publish(channelName, JSON.stringify({ event: event, data: data }));
+	if (nconf.get('isCluster') === 'false') {
+		var EventEmitter = require('events');
+		pubsub = new EventEmitter();
+		pubsub.publish = pubsub.emit.bind(pubsub);
+	} else if (nconf.get('redis')) {
+		pubsub = require('./database/redis').pubsub;
```

Comment:
when you have mongodb as datastore and redis as sessionstore, this always returns undefined, I fixed it on the develop branch.

Reply:
Thanks, cherry-picked.